cimgen/languages/modernpython/utils/base.py

```diff
@@ -132,38 +151,217 @@ def cgmes_attributes_in_profile(self, profile: BaseProfile | None) -> dict[str,
+
+            root = self._add_attribute_to_etree(attributes=profile_attributes, root=root, nsmap=nsmap)
+        return root
+
+    def _get_attributes_to_export(self, profile_to_export: BaseProfile) -> dict:
+        attributes_to_export = {}
+        attributes_in_profile = self.cgmes_attributes_in_profile(profile_to_export)
+        for key, attribute in attributes_in_profile.items():
+            if attribute["attribute_main_profile"] == profile_to_export:
+                attributes_to_export[key] = attribute
+        attributes_to_export = self._remove_empty_attributes(attributes_to_export)
+        return attributes_to_export
+
+    @staticmethod
+    def _remove_empty_attributes(attributes: dict) -> dict:
+        for key, attribute in list(attributes.items()):
+            # Remove empty attributes
+            if attribute["value"] in [None, "", []]:
+                del attributes[key]
+            elif attribute.get("attribute_class") and attribute["attribute_class"] == "Boolean":
+                attribute["value"] = str(attribute["value"]).lower()
+        return attributes
+
+    @staticmethod
+    def _add_attribute_to_etree(attributes: dict, root: etree.Element, nsmap: dict) -> etree.Element:
+        rdf_namespace = f"""{{{nsmap["rdf"]}}}"""
+        for field_name, attribute in attributes.items():
+            # add all attributes relevant to the profile as SubElements
+            attr_namespace = attribute["namespace"]
+            element_name = f"{{{attr_namespace}}}{field_name}"
+
+            if attribute["is_class_attribute"]:
+                # class_attributes are exported as rdf: resource #mRID_of_target
+                element = etree.SubElement(root, element_name)
+                element.set(rdf_namespace + "resource", "#" + attribute["value"])
+            elif attribute["is_enum_attribute"]:
+                element = etree.SubElement(root, element_name)
+                element.set(rdf_namespace + "resource", nsmap["cim"] + attribute["value"])
+            else:
+                element = etree.SubElement(root, element_name)
+                element.text = str(attribute["value"])
+        return root
+
     def __str__(self) -> str:
         """Returns the string representation of this resource."""
         return "\n".join([f"{k}={v}" for k, v in sorted(self.to_dict().items())])
 
+    def _parse_xml_fragment(self, xml_fragment: str) -> dict:
+        """parses an xml fragment into a dict defining the class attributes
+
+        Args:
+            xml_fragment (str): xml string defining an instance of the current class
+
+        Returns:
+            attribute_dict (dict): a dictionnary of attributes to create/update the class instance
+        """
+        attribute_dict = {}
+        xml_tree = etree.fromstring(xml_fragment)
+
+        # raise an error if the xml does not describe the expected class
+        if not xml_tree.tag.endswith(self.resource_name):
```

Comment:
```suggestion
        if not str(xml_tree.tag).endswith(self.resource_name):
```
xml_tree.tag is not in every case a string.